In [1]:
import pandas as pd
import numpy as np

# Task 1.1

In [2]:
data = pd.read_csv("./HDFCBANK.csv")

In [3]:
print(data.shape)
data.head()

(609, 15)


Symbol Series         Date  Prev Close  Open Price  High Price  \
0  HDFCBANK     EQ  15-May-2017     1551.65      1550.0      1557.2   
1  HDFCBANK     IL  15-May-2017     1567.00      1583.0      1583.0   
2  HDFCBANK     EQ  16-May-2017     1553.40      1555.0      1562.0   
3  HDFCBANK     EQ  17-May-2017     1559.65      1558.8      1565.4   
4  HDFCBANK     IL  17-May-2017     1583.00      1600.0      1600.0   

   Low Price      Last Price  Close Price  Average Price  \
0    1542.25         1553.00      1553.40        1550.73   
1    1583.00               -      1583.00        1583.00   
2    1548.25         1561.85      1559.65        1555.11   
3    1551.50         1557.50      1557.15        1557.51   
4    1600.00               -      1600.00        1600.00   

   Total Traded Quantity      Turnover  No. of Trades  Deliverable Qty  \
0                 701195  1.087368e+09          15104           309282   
1                 342000  5.413860e+08              6           342000   
2                 897594  1.395857e+09          21073           347734   
3                 868118  1.352103e+09          32732           367931   
4                  50000  8.000000e+07              1            50000   

   % Dly Qt to Traded Qty  
0                   44.11  
1                  100.00  
2                   38.74  
3                   42.38  
4                  100.00

In [4]:
# Arrange all rows by value in index Series
data = data.set_index('Series')
# Drop all the rows with IL in Series
data = data.drop('IL')

In [5]:
# Reseting the index the table
data = data.reset_index()

In [6]:
print(data.shape)
data.head()

(495, 15)


Series    Symbol         Date  Prev Close  Open Price  High Price  \
0     EQ  HDFCBANK  15-May-2017     1551.65      1550.0      1557.2   
1     EQ  HDFCBANK  16-May-2017     1553.40      1555.0      1562.0   
2     EQ  HDFCBANK  17-May-2017     1559.65      1558.8      1565.4   
3     EQ  HDFCBANK  18-May-2017     1557.15      1549.0      1561.5   
4     EQ  HDFCBANK  19-May-2017     1557.10      1559.5      1569.9   

   Low Price      Last Price  Close Price  Average Price  \
0    1542.25         1553.00      1553.40        1550.73   
1    1548.25         1561.85      1559.65        1555.11   
2    1551.50         1557.50      1557.15        1557.51   
3    1548.10         1557.50      1557.10        1556.99   
4    1545.25         1568.05      1561.25        1556.93   

   Total Traded Quantity      Turnover  No. of Trades  Deliverable Qty  \
0                 701195  1.087368e+09          15104           309282   
1                 897594  1.395857e+09          21073           347734   
2                 868118  1.352103e+09          32732           367931   
3                 776820  1.209500e+09          13648           518101   
4                 713534  1.110926e+09          18458           296680   

   % Dly Qt to Traded Qty  
0                   44.11  
1                   38.74  
2                   42.38  
3                   66.70  
4                   41.58

In [7]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 495 entries, 0 to 494
Data columns (total 15 columns):
Series                    495 non-null object
Symbol                    495 non-null object
Date                      495 non-null object
Prev Close                495 non-null float64
Open Price                495 non-null float64
High Price                495 non-null float64
Low Price                 495 non-null float64
Last Price                495 non-null object
Close Price               495 non-null float64
Average Price             495 non-null float64
Total Traded Quantity     495 non-null int64
Turnover                  495 non-null float64
No. of Trades             495 non-null int64
Deliverable Qty           495 non-null int64
% Dly Qt to Traded Qty    495 non-null float64
dtypes: float64(8), int64(3), object(4)
memory usage: 58.1+ KB


In [8]:
data.describe()

Prev Close   Open Price   High Price    Low Price  Close Price  \
count   495.000000   495.000000   495.000000   495.000000   495.000000   
mean   1953.787677  1957.834747  1972.316768  1943.926768  1958.558889   
std     189.658475   178.021896   179.491320   175.802014   177.283726   
min     470.000000  1549.000000  1557.200000  1542.250000  1553.400000   
25%    1834.650000  1836.725000  1849.950000  1822.300000  1838.100000   
50%    1952.400000  1955.000000  1972.200000  1936.600000  1956.000000   
75%    2098.125000  2100.500000  2112.500000  2085.500000  2099.925000   
max    2367.950000  2364.200000  2378.000000  2356.500000  2367.950000   

       Average Price  Total Traded Quantity      Turnover  No. of Trades  \
count     495.000000           4.950000e+02  4.950000e+02     495.000000   
mean     1958.440384           2.216954e+06  4.466206e+09   72178.317172   
std       177.490791           2.057706e+06  4.456149e+09   47925.252706   
min      1550.730000           1.202980e+05  2.521687e+08       1.000000   
25%      1839.230000           1.142748e+06  2.117527e+09   31724.500000   
50%      1957.170000           1.772797e+06  3.477097e+09   61019.000000   
75%      2098.430000           2.644762e+06  5.462882e+09  104439.500000   
max      2370.960000           2.703218e+07  6.067489e+10  261775.000000   

       Deliverable Qty  % Dly Qt to Traded Qty  
count     4.950000e+02              495.000000  
mean      1.337857e+06               59.155818  
std       1.257774e+06               10.453805  
min       6.900200e+04               26.220000  
25%       6.498170e+05               52.465000  
50%       9.981860e+05               59.830000  
75%       1.642846e+06               66.305000  
max       1.391462e+07              100.000000

In [9]:
data.tail()

Series    Symbol         Date  Prev Close  Open Price  High Price  \
490     EQ  HDFCBANK  07-May-2019     2328.45     2351.90     2358.00   
491     EQ  HDFCBANK  08-May-2019     2322.35     2322.55     2323.80   
492     EQ  HDFCBANK  09-May-2019     2307.25     2300.00     2300.00   
493     EQ  HDFCBANK  10-May-2019     2290.90     2291.35     2324.90   
494     EQ  HDFCBANK  13-May-2019     2296.50     2296.05     2313.35   

     Low Price      Last Price  Close Price  Average Price  \
490    2318.00         2320.10      2322.35        2338.90   
491    2288.00         2309.00      2307.25        2304.51   
492    2271.60         2292.00      2290.90        2287.67   
493    2291.35         2296.50      2296.50        2306.93   
494    2280.00         2287.30      2287.20        2297.73   

     Total Traded Quantity      Turnover  No. of Trades  Deliverable Qty  \
490                2481051  5.802928e+09         114540          1326590   
491                3016052  6.950518e+09         101965          1942147   
492                2256931  5.163124e+09         129719          1470611   
493                2648550  6.110008e+09          79307          1542118   
494                2767908  6.359892e+09         145875          1652663   

     % Dly Qt to Traded Qty  
490                   53.47  
491                   64.39  
492                   65.16  
493                   58.22  
494                   59.71

# Task 1.2

In [21]:
data_90_days = data.iloc?

In [10]:
data_90_days = data.iloc[-90:]
print(data_90_days.shape)

(90, 15)


In [29]:
print(data_90_days['Close Price'].max())
print(data_90_days['Close Price'].min())
print(data_90_days['Close Price'].mean())

2367.95
2034.7
2188.070555555556


# Task 1.3

In [39]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 495 entries, 0 to 494
Data columns (total 15 columns):
Series                    495 non-null object
Symbol                    495 non-null object
Date                      495 non-null datetime64[ns]
Prev Close                495 non-null float64
Open Price                495 non-null float64
High Price                495 non-null float64
Low Price                 495 non-null float64
Last Price                495 non-null object
Close Price               495 non-null float64
Average Price             495 non-null float64
Total Traded Quantity     495 non-null int64
Turnover                  495 non-null float64
No. of Trades             495 non-null int64
Deliverable Qty           495 non-null int64
% Dly Qt to Traded Qty    495 non-null float64
dtypes: datetime64[ns](1), float64(8), int64(3), object(3)
memory usage: 58.1+ KB


In [41]:
data['Date'] = data['Date'].astype('datetime64')

In [38]:
print(max(data['Date'])-min(data['Date']))

728 days 00:00:00


# Task 1.4

In a separate array , calculate the monthwise VWAP (Volume Weighted Average Price ) of the stock.
  ( VWAP = sum(price*volume)/sum(volume) ) 
To know more about VWAP , visit - VWAP definition 
{Hint : Create a new dataframe column ‘Month’.
The values for this column can be derived from the ‘Date” column by using appropriate pandas functions.
Similarly, create a column ‘Year’ and initialize it. Then use the 'groupby()' function by month and year.
Finally, calculate the vwap value for each month (i.e. for each group created).